### PR TITLE
feat: search filter checkbox - LESQ-786

### PR DESCRIPTION
**Pull request by JBR90**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

## Link to the design doc
https://www.figma.com/file/YcWQMMhHPVVmc47cHHEEAl/Oak-Design-Kit?node-id=3037%3A6717&mode=dev

## A link to the component in the deployment preview
https://deploy-preview-148--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-teacher-oaksearchfiltercheckbox--docs

## ACs
https://www.notion.so/oaknationalacademy/I-want-a-search-filter-checkbox-in-teacher-component-library-622c28573e6348398fb3c7be50270d24

### CODE REVIEW

**Comment by JBR90** [comment]
@simonh-oak I'm wondering now if this could be a more generic molecule now, OakCheckBoxTile or something ?

File: src/components/organisms/teacher/OakSearchFilterCheckBox/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from "./OakSearchFilterCheckBox";
```
Comment (njblore) [inline src/components/organisms/teacher/OakSearchFilterCheckBox/index.ts:1]: should we export the props type? unsure if we're doing that elsewhere in the library but feels useful

File: src/components/organisms/pupil/OakQuizCheckBox/OakQuizCheckBox.tsx [hunk 2]
```diff
@@ -204,7 +200,8 @@ export const OakQuizCheckBox = (props: OakQuizCheckBoxProps) => {
         isHighlighted ? "border-decorative5-stronger" : "transparent"
       }
     >
-      <StyledInternalCheckBoxLabelHoverDecor
+      <InternalCheckBoxLabelHoverDecor
```
Comment (kimon-satan) [inline src/components/organisms/pupil/OakQuizCheckBox/OakQuizCheckBox.tsx:203]: Is this change intentional ., I'm worried about regressions on the pupil journey.

**Review by kimon-satan — CHANGES_REQUESTED** [review]
Nice one !

File: src/components/organisms/teacher/OakSearchFilterCheckBox/OaKSearchFilterCheckBox.test.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
+import React, { createRef } from "react";
```
[the remark this answers is not in the document]
Comment (JBR90) [reply src/components/organisms/teacher/OakSearchFilterCheckBox/OaKSearchFilterCheckBox.test.tsx:1]: good spot!

File: src/components/organisms/teacher/OakSearchFilterCheckBox/index.ts (hunk 1 above)
Comment (JBR90) [reply src/components/organisms/teacher/OakSearchFilterCheckBox/index.ts:1]: `export * `- Will that not export the prop types ?

Comment (njblore) [reply src/components/organisms/teacher/OakSearchFilterCheckBox/index.ts:1]: oh! yeah it will 👀

**Review by njblore — APPROVED** [review]

File: src/components/organisms/pupil/OakQuizCheckBox/OakQuizCheckBox.tsx (hunk 2 above)
Comment (kimon-satan) [reply src/components/organisms/pupil/OakQuizCheckBox/OakQuizCheckBox.tsx:203]: Ah got it. That makes sense.

**Review by kimon-satan — APPROVED** [review]

**Comment by Monika-Koziol** [comment]
@JBR90 a few small changes to the designs:

- [ ] Can we use a thinner border border-solid-S
- [ ] Can we use 'Sentence case' for all subject names 
- [ ] Can the hover state have underline the text
